tests/Doctrine/Tests/ORM/Utility/IdentifierFlattenerDateTimeImmutableIdTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Utility;
+
+use DateTimeImmutable;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Mapping\Driver\AttributeDriver;
+use Doctrine\Tests\Models\DateTimeCompositeKey\Article;
+use Doctrine\Tests\Models\DateTimeCompositeKey\ArticleAudit;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+use function dirname;
+use function sleep;
+
+/**
+ * Test the IdentifierFlattener utility class
+ *
+ * @requires PHP 8.1
+ * @covers \Doctrine\ORM\Utility\IdentifierFlattener
+ */
+class IdentifierFlattenerDateTimeImmutableIdTest extends OrmFunctionalTestCase
+{
+    /**
+     * @var EntityRepository
+     */
+    private $articleRepository;
+
+    /**
+     * @var EntityRepository
+     */
+    private $articleAuditRepository;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->_em = $this->getEntityManager(null, new AttributeDriver([dirname(__DIR__, 2) . '/Models/Enums']));
+        $this->createSchemaForModels(
+            Article::class,
+            ArticleAudit::class
+        );
+        $this->articleRepository      = $this->_em->getRepository(Article::class);
+        $this->articleAuditRepository = $this->_em->getRepository(ArticleAudit::class);
+    }
+
+    /** @group utilities */
+    public function testFlattenIdentifierWithDateTimeId(): void
+    {
+        $article = new Article('Some title');
+        $article->changeTitle('New title');
+        sleep(1);
```

Comment:
Please find a way to avoid this. Maybe define a class that extends `DateTimeImmutable` and makes sure to add 1 second when the constructor is called, I don't know.

BTW, doesn't this point at a weakness of having time part of the composite key? What happens in high throughput scenarios? People might end up with entities with the same ID, might they not? I'm starting to have some doubts about this…

Reply:
It's not that I want to tell people what's wrong or right, but I would like to avoid people opening bug reports, have us help them debug only to find they are hitting this limitation. That's why I'm weary of code that works only if people know what they are doing.